compiler/rustc_interface/src/interface.rs

```diff
@@ -215,6 +230,116 @@ pub fn parse_check_cfg(handler: &EarlyErrorHandler, specs: Vec<String>) -> Check
                                 } else {
                                     expected_error();
                                 }
+                            } else if meta_item.has_name(sym::cfg) {
+                                old_syntax = Some(false);
+
+                                let mut names = Vec::new();
+                                let mut values: FxHashSet<_> = Default::default();
+
+                                let mut any_specified = false;
+                                let mut values_specified = false;
+                                let mut values_any_specified = false;
+
+                                for arg in args {
+                                    if arg.is_word() && let Some(ident) = arg.ident() {
+                                        if values_specified {
+                                            error!("`cfg()` names cannot be after values");
+                                        }
+                                        names.push(ident);
+                                    } else if let Some(args) = arg.meta_item_list()
+                                        && arg.has_name(sym::any)
```

Comment:
Could you change the condition order to be like `arg.has_name(sym::any) && let Some(args) = arg.meta_item_list()`?
(Here and in all places below.)

We don't need the list if the argument is not `any`, so this version is easier to read, and also probably faster.